src/topology/continuous_function/algebra.lean

```diff
@@ -71,32 +71,33 @@ the structure of a group.
 section subtype
 
 @[to_additive]
-instance continuous_submonoid (α : Type*) (β : Type*) [topological_space α] [topological_space β]
-  [monoid β] [has_continuous_mul β] : is_submonoid { f : α → β | continuous f } :=
-{ one_mem := @continuous_const _ _ _ _ 1,
-  mul_mem := λ f g fc gc, continuous.comp
-  has_continuous_mul.continuous_mul (continuous.prod_mk fc gc : _) }
+def continuous_submonoid (α : Type*) (β : Type*) [topological_space α] [topological_space β]
+  [monoid β] [has_continuous_mul β] : submonoid (α → β) :=
+{ carrier := { f : α → β | continuous f },
+  one_mem' := @continuous_const _ _ _ _ 1,
+  mul_mem' := λ f g fc gc, continuous.comp
+    has_continuous_mul.continuous_mul (continuous.prod_mk fc gc : _) }
 
 @[to_additive]
-instance continuous_subgroup (α : Type*) (β : Type*) [topological_space α] [topological_space β]
-  [group β] [topological_group β] : is_subgroup { f : α → β | continuous f } :=
-{ inv_mem := λ f fc, continuous.comp (@topological_group.continuous_inv β _ _ _) fc,
+def continuous_subgroup (α : Type*) (β : Type*) [topological_space α] [topological_space β]
+  [group β] [topological_group β] : subgroup (α → β) :=
+{ inv_mem' := λ f fc, continuous.comp (@topological_group.continuous_inv β _ _ _) fc,
   ..continuous_submonoid α β, }.
 
 @[to_additive]
 instance continuous_monoid {α : Type*} {β : Type*} [topological_space α] [topological_space β]
   [monoid β] [has_continuous_mul β] : monoid { f : α → β | continuous f } :=
-subtype.monoid
+(continuous_submonoid α β).to_monoid
 
 @[to_additive]
 instance continuous_group {α : Type*} {β : Type*} [topological_space α] [topological_space β]
   [group β] [topological_group β] : group { f : α → β | continuous f } :=
-subtype.group
+(continuous_subgroup α β).to_group
 
 @[to_additive]
 instance continuous_comm_group {α : Type*} {β : Type*} [topological_space α] [topological_space β]
   [comm_group β] [topological_group β] : comm_group { f : α → β | continuous f } :=
-@subtype.comm_group _ _ _ (continuous_subgroup α β) -- infer_instance doesn't work?!
+(continuous_subgroup α β).to_comm_group
```

Comment:
I would be in favour of deleting these altogether, and the similar constructions for rings.  The comment at the start of this file,

> Instances are present both in the case of the subtype of continuous functions and the type of continuous bundled functions. Both implementations have advantages and disadvantages, but many experienced people in Zulip have expressed a preference towards continuous bundled maps, so when there is no particular reason to use the subtype, continuous bundled functions should be used for the sake of uniformity.

seems strange to me every time I read it -- it gives no reason why the subtype version should exist at all!  And if mathlib has the structure as a bundled subgroup, subring, etc, it will be easy for anyone who truly needs it to get the group, ring, etc instances.

Reply:
(But @ADedecker, you should probably wait to hear other people's opinions on this before following mine!)

Reply:
Just to double check : you want to remove the `monoid`, `group`, `ring` ... instances for the subset version, while keeping the corresponding `subgroup`s since they can't be expressed with the bundled version ? If so, I absolutely agree with you. Should we check for a consensus before ?

Reply:
Yes, exactly!